### ISSUE TITLE
Use library along with global pipes

### ISSUE THREAD

**Issue by YannisSerli** [Adrinalin4ik/Nestjs-Graphql-Tools#22]
In the release 0.7.17, when we follow the readme adding @Filter() parameter with type of `Brackets` from typeorm library, an error is throw: `where.whereFactory is not a function`.

Changing the type `Brackets` into `any` solve the problems.

**Comment by Adrinalin4ik** [comment]
@YannisSerli can you give more information please. I want to know typeorm and  graphql versions. Also attach some code snippet.

**Comment by YannisSerli** [comment]
The typeorm and graphql versions:

`"@nestjs/typeorm": "^9.0.1"`
`"@nestjs/graphql": "^10.1.7"`

A code snippet that doesn't work:

```js
  /**
   * Find all persons
   * @returns all persons
   */
  @Query(() => [Person], { name: 'persons' })
  @UseGuards(GqlAuthGuard, CoachResearcherGuard)
  @GraphqlFilter()
  @GraphqlSorting()
  async findAll(
    @Filter(() => Person) filter: Brackets,
    @Sorting(() => Person) sorting: SortArgs<Person>,
    @Paginator() paginator,
  ): Promise<Person[]> {
    const query = repository.createQueryBuilder();
    query.where(filter);
    query.orderBy(sorting);
    query.offset(paginator.page * paginator.per_page).limit(paginator.per_page);
    return query.getMany();
  }
```

**Comment by Adrinalin4ik** [comment]
@YannisSerli Can you please look at [src/entities/user/user.resolver](https://github.com/Adrinalin4ik/Nestjs-Graphql-Tools/blob/master/src/entities/user/user.resolver.ts). There's no guards here. Probably UseGuards overrides all the rest guards. Can you please try to move you UseGuards  after @GraphqlSorting()?

**Comment by Adrinalin4ik** [comment]
Another thing what you can do is to change the logic in GqlAuthGuard and CoachResearcherGuard. You need to call action and spread all parameters to it. Also, make sure that you don't override initial method there, because in this case you will loose the context.

**Comment by YannisSerli** [comment]
I moved UseGuards after @GraphqlSorting and even removed the UseGuards but the problem stay the same.

**Comment by YannisSerli** [comment]
I think the problem comes from `lib/filters/query.builder.ts` line 19: `if (parameters && 'whereFactory' in parameters) return parameters;`

Because using Brackets the `if` condition became true but `whereFactory` is undefined.

**Comment by Adrinalin4ik** [comment]
That's super strange and if you run example that inside of src folder it will work. You mentioned that having any type instead of bracket makes it works. At the same time types don't exist in the final bundle as it compiles to js. How it can break?

**Comment by Adrinalin4ik** [comment]
Closing due to inactivity.

**Comment by KyDenZ** [comment]
I've had the same problem since I added validations to bootstrap:

`app.useGlobalPipes(new ValidationPipe({ transform: true }));`

**Comment by Adrinalin4ik** [comment]
Oh, it looks like global pipes override results of controllers pipes. I'll take a look. For now, you can try to use pipes locally wrapping the methods in the controller. The first pipe should be yours, and after all the following decorators that comes fom the library.

**Comment by Adrinalin4ik** [comment]
I was able to reproduce the issue, working on it.

**Comment by Adrinalin4ik** [comment]
@KyDenZ @YannisSerli I hope it was fixed in the version 0.8.0 https://github.com/Adrinalin4ik/Nestjs-Graphql-Tools/releases/tag/0.8.0
